### DISCUSSION TITLE
Image Compression: Imgbot GitHub app vs built-in Optimize plugin

### DISCUSSION THREAD

**Discussion by santaimpersonator** [squidfunk/mkdocs-material#7569] [Q&A]
Has anyone tested the Imgbot Github app against the built-in Optimize plugin, from the MkDocs-Material theme?

|   | [Imgbot GitHub App](https://github.com/marketplace/imgbot) | [Built-in Optimize Plugin](https://squidfunk.github.io/mkdocs-material/plugins/optimize/) |
| :-----: |:------ | :------ |
| Pros | Provides image compression for multiple file types<br>Utilize either a lossy or lossless compression<br>Open-source and verified GitHub app<br><ul><li>Free for public GitHub repositories</li></ul> | Only compresses images when copying them to the [`site` directory](https://www.mkdocs.org/user-guide/configuration/#site_dir)<br>Only runs on builds for documentation, in a specific repository |
| Cons | Runs across all repositories<br>Compresses images directly in the repositories | Only provides image compression for `*.png` and `*.jpg` files<br>Insiders only feature |

**Comment by squidfunk** [comment]
To add to that: you can actually [combine it with the privacy plugin](https://squidfunk.github.io/mkdocs-material/setup/ensuring-data-privacy/#built-in-privacy-plugin) to store images outside of the repository (which VCS like git don't particularly like due to their binary nature), creating a workflow that downloads, optimizes, and inlines those images with your documentation.

> This option makes the [built-in privacy plugin](https://squidfunk.github.io/mkdocs-material/plugins/privacy/) an excellent choice for when you want to host assets like images outside of your git repository in another location to keep them fresh and your repository lean.

> Additionally, as of [insiders-4.30.0](https://squidfunk.github.io/mkdocs-material/insiders/changelog/#4.30.0), the built-in privacy plugin was entirely rewritten and now works perfectly with the [built-in optimize plugin](https://squidfunk.github.io/mkdocs-material/plugins/optimize/), which means that external assets can be passed through the same optimization pipeline as the rest of your documentation. This means you can store and edit unoptimized files outside of your repository, and let both plugins built a highly optimized site for you.
>
> If you want to implement separate pipelines, i.e., optimize some images differently from others or exclude some images from downloading, you can use multiple instances of the [built-in privacy plugin](https://squidfunk.github.io/mkdocs-material/plugins/privacy/).
>
Additionally, you can run the optimize plugin in constrained environments without internet connection, no third-party service needed. Other than that, we'll provide more possibilities for optimization in the future, but we currently have other priorities. Additionally, the optimize plugin will soon be released to the public version 🤟
